optaplanner-core/src/main/java/org/optaplanner/core/config/util/ConfigUtils.java

```diff
@@ -274,13 +274,18 @@ public static int resolveThreadPoolSizeScript(String propertyName, String script
 
     public static List<Class<?>> getAllAnnotatedLineageClasses(Class<?> bottomClass,
             Class<? extends Annotation> annotation) {
+        if (bottomClass == null) {
+            throw new IllegalArgumentException("The inspected class (bottomClass) cannot be null");
```

Comment:
error message should include relevant runtime information, in this case the bottomClass's name:

`throw new IllegalArgumentException("The bottomClass (" + bottomClass + ") cannot be null");`

Reply:
But.... ```bottomClass``` is ```null``` so this'll show "The bottomClass (null) cannot be null"!!

Reply:
What @ge0ffrey is suggesting here is a standard pattern (in optaplanner) of constructing exception messages whose purpose is to explain a programmer's mistake. Generally it's `The <description> <variableName> (<its value>) is invalid for some reason.`. I agree that including both the invalid value and the variable name is helpful when debugging and reading the code [1]. However in case of null checking this pattern starts to provide redundant information, exactly as @manstis pointed out. Besides that, some tools like FindBugs will report this as dereferencing null pointer.

[1] For example: `The argument inputArg (-9) cannot be negative because square root is not defined for negative numbers (when the codomain is real).`